src/pypi_mapping/custom_pypi_mapping.rs

```diff
@@ -0,0 +1,129 @@
+use miette::{Context, IntoDiagnostic};
+use rattler_conda_types::{PackageUrl, RepoDataRecord};
+use reqwest_middleware::ClientWithMiddleware;
+use std::{collections::HashMap, sync::Arc};
+
+use async_once_cell::OnceCell;
+
+use crate::pypi_mapping::MappingLocation;
+
+use super::{prefix_pypi_name_mapping, MappingMap, Reporter};
+
+pub async fn fetch_custom_mapping(
+    client: &ClientWithMiddleware,
+    mapping_url: &MappingMap,
+) -> miette::Result<&'static HashMap<String, HashMap<String, String>>> {
+    static MAPPING: OnceCell<HashMap<String, HashMap<String, String>>> = OnceCell::new();
+    MAPPING
+        .get_or_try_init(async {
+            let mut mapping_url_to_name: HashMap<String, HashMap<String, String>> =
+                Default::default();
+
+            for (name, url) in mapping_url.iter() {
+                // Fetch the mapping from the server or from the local
+
+                match url {
+                    MappingLocation::Url(url) => {
+                        let response = client
+                            .get(url.clone())
+                            .send()
+                            .await
+                            .into_diagnostic()
+                            .context("failed to download pypi mapping from custom location")?;
```

Comment:
I tested what happens if you give a broken URL but then this doesn't get shown but then next one does. Could you make sure that it fails on downloading, and that it adds atleast also adds the url to the error so it helps the user find the cultprit? Now it gives this error:
```
  × failed to parse pypi name mapping
  ├─▶ error decoding response body: invalid type: integer `404`, expected a map at line 1 column 3
  ╰─▶ invalid type: integer `404`, expected a map at line 1 column 3
```

Reply:
<img width="1019" alt="Screenshot 2024-04-08 at 14 56 24" src="https://github.com/prefix-dev/pixi/assets/25200469/2a08184a-c215-4f36-8deb-4364e8b982fa">
change the logic so know it will throw following error if requested mapping could not be located

Reply:
and following error if json is bad formatted ( for example, could not be able to serialized in our struct )
<img width="894" alt="Screenshot 2024-04-08 at 14 58 36" src="https://github.com/prefix-dev/pixi/assets/25200469/a8757b24-c0f9-40a1-a7ed-d4535e863257">